src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/DevCanvasResultSourceService.cs

```diff
@@ -0,0 +1,384 @@
+        /// </summary>
+        /// <param name="state">A <see cref="RolledUpCacheItem"/> object.</param>
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        private async void QueryInsights(object state)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+        {
+            try
+            {
+                string absoluteFilePath = (string)state;
+                int insightsCount = 0;
+
+                // This will contain the message we'll display in the output window once the query is complete (and after we release the lock on cacheItem).
+                StringBuilder resultMessage = new StringBuilder();
+
+                // TODO 
+                // See if the file is part of a git repo / source depot base
+                SourceControlType sourceControlType = SourceControlType.Unknown;
+                string gitRepoRoot = await gitExe.GetRepoRootAsync(absoluteFilePath);
+                string server = "";
+                string project = "";
+                string repo = "";
+                string branch = "";
+
+                if (gitRepoRoot != null)
+                {
+                    sourceControlType = SourceControlType.Git;
+                    string repoUri = await gitExe.GetRepoUriAsync(absoluteFilePath);
+                    Util.ParseGitUrl(repoUri, out server, out project, out repo);
+                    if (string.IsNullOrWhiteSpace(server) ||
+                        string.IsNullOrWhiteSpace(project) ||
+                        string.IsNullOrWhiteSpace(repo))
+                    {
+                        sourceControlType = SourceControlType.Unknown;
+                    }
+                    else
+                    {
+                        branch = await gitExe.GetCurrentBranchAsync(absoluteFilePath);
+                    }
+                }
+                else
+                {
+                    bool sdExists = Util.IsSourceDepot(absoluteFilePath, out gitRepoRoot);
+                    if (sdExists)
+                    {
+                        sourceControlType = SourceControlType.SourceDepot;
+                    }
+                }
+
+                if (sourceControlType == SourceControlType.Unknown)
+                {
+                    return;
+                }
+
+                string repoRootedFilePath = GetRepoRootedFilePath(gitRepoRoot, absoluteFilePath);
+
+                // if it is part of a code base, construct a DevCanvasVersionControlDetails with appropriate details 
+                DevCanvasVersionControlDetails vcDetails = new DevCanvasVersionControlDetails(sourceControlType, server, project, repo, branch);
+
+                // Get the generators that we should query for this cache item.
+                List<DevCanvasGeneratorInfo> allGenerators = await accessor.GetGeneratorsAsync();
+
+                // Now query the server for insights.
+                Trace.WriteLine($"Querying for insights for {absoluteFilePath}");
```

Comment:
Confirming that 'Trace.WriteLine()' goes to SARIF Viewer output pane?

Reply:
*If it does, do we want to start a paradigm to have plugin information prefixed in some way to make it more obvious who is looking this?

Reply:
@EasyRhinoMSFT Is this something we would want to standardize across the ADO and Ghasdo plugins?

Reply:
For now I've put a "DevCanvas" tag before all of the traces from devcanvas. We can revisit this in the future if Chris and co want to I think.